### PR TITLE
[PDR-451] Add new null/skip value to UBR calculations.

### PR DESCRIPTION
**Pull request by robabram**
Full suite of unittests for UBR calculator code.

## Description of changes/additions
Add null/skip value to UBR calculations, UBR is no longer a Yes or No calculation.  A Null or 'PMI_Skip' value is now a new calculation result value.
Added a full suite of unittests for UBR calculator code.

## Tests
- [x] unit tests

### CODE REVIEW

File: rdr_service/resource/calculators/participant_ubr.py [hunk 1]
```diff
@@ -48,8 +62,10 @@ def ubr_gender_identity(birth_sex, gender_ident, gender_ident_closer):
                         survey.
         :param gender_ident_closer: Comma delimited str of answer codes to GenderIdentity_SexualityCloserDescription
                         question in "TheBasics" survey.
-        :return: 1 if UBR else 0
+        :return: UBRValueEnum
         """
+        if gender_ident is None and gender_ident_closer is None:
```
Comment (pegbertsch) [inline rdr_service/resource/calculators/participant_ubr.py:67]: Just confirming that the PMI_Skip answer code is not a factor for these (along with checking for None)?

File: tests/resource_tests/calculator_tests/test_ubr_calculator.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        consent_2020 = RURAL_2020_CUTOFF  # Hit 2020 version of rural zipcodes.
+
+        zip_2014 = '35052'  # Only in 2014 file.
+        zip_2020 = '35063'  # Only in 2020 file.
+        zip_both = '72529'  # In both the 2014 file and 2020 file.
+        zip_none = '58493'  # Zip code not in either file.
+
+        # Test with Null and PMI_Skip values
+        self.assertEqual(self.ubr.ubr_geography(consent_2014, None), UBRValueEnum.NullSkip)
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, None), UBRValueEnum.NullSkip)
```
Comment (pegbertsch) [inline tests/resource_tests/calculator_tests/test_ubr_calculator.py:204]: Is PMI_Skip a code that needs to be tested as the 2nd param (per comment), or was that a cut/pasted comment from another test?

File: tests/resource_tests/calculator_tests/test_ubr_calculator.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        # Test zip code in both files for UBR
+        self.assertEqual(self.ubr.ubr_geography(consent_2014, zip_both), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, zip_both), UBRValueEnum.UBR)
+
+        # Test for RBR value
+        self.assertEqual(self.ubr.ubr_geography(consent_2014, zip_none), UBRValueEnum.RBR)
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, zip_none), UBRValueEnum.RBR)
+
+        # Bad or unknown value will default to RBR.
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, '123-abc'), UBRValueEnum.RBR)
```
Comment (pegbertsch) [inline tests/resource_tests/calculator_tests/test_ubr_calculator.py:225]: Is it worth adding a case with a ZIP+4 response ?

**Review by pegbertsch — APPROVED** [review]
Approved, assuming there is no need to incorporate checks for PMI_Skip when looking at GenderIdentity/GenderIdentity_SexualityCloserDescription (it would make sense to me that PMI_Skip isn't an option based on survey structure and branching logic)

File: rdr_service/resource/calculators/participant_ubr.py (hunk 1 above)
Comment (robabram) [reply rdr_service/resource/calculators/participant_ubr.py:67]: The Skip value get tested in the code later when the two value sets are combined.

File: tests/resource_tests/calculator_tests/test_ubr_calculator.py (hunk 2 above)
Comment (robabram) [reply tests/resource_tests/calculator_tests/test_ubr_calculator.py:204]: Yes, I just copied that comment. There are no PMI_Skip zip code values in the data, but there are Nulls.

File: tests/resource_tests/calculator_tests/test_ubr_calculator.py (hunk 3 above)
Comment (robabram) [reply tests/resource_tests/calculator_tests/test_ubr_calculator.py:225]: Yes, that is a good suggestion, I'll add a couple of test cases.

**Review by wangy70 — APPROVED** [review]

**Review by pegbertsch — APPROVED** [review]

**Pull request merged**